packages/vscode-extension/src/common/Project.ts

```diff
@@ -27,35 +27,66 @@ export type ToolsState = {
   [key: string]: ToolState;
 };
 
-export type ProjectState =
-  | ({
-      status:
-        | "starting"
-        | "running"
-        | "bootError"
-        | "bundlingError"
-        | "debuggerPaused"
-        | "refreshing";
-    } & ProjectStateCommon)
-  | ProjectStateBuildError;
-
-type ProjectStateCommon = {
-  previewURL: string | undefined;
-  selectedDevice: DeviceInfo | undefined;
-  initialized: boolean;
-  previewZoom: ZoomLevelType | undefined; // Preview specific. Consider extracting to different location if we store more preview state
+export type BuildErrorDescriptor = {
+  message: string;
+  platform: DevicePlatform;
+  buildType: BuildType | null;
+};
+
+export type ProfilingState = "stopped" | "profiling" | "saving";
+
+export type NavigationHistoryItem = {
+  displayName: string;
+  id: string;
+};
+
+export type DeviceSessionStatus =
+  | "starting"
+  | "running"
+  | "bootError"
+  | "bundlingError"
+  | "debuggerPaused"
+  | "refreshing"
+  | "buildError";
+
+export type DeviceSessionState = {
+  status: DeviceSessionStatus;
   startupMessage: StartupMessage | undefined;
   stageProgress: number | undefined;
+  buildError: BuildErrorDescriptor | undefined;
+  selectedDevice: DeviceInfo | undefined;
+  previewURL: string | undefined;
+  profilingReactState: ProfilingState;
+  profilingCPUState: ProfilingState;
+  navigationHistory: NavigationHistoryItem[];
+  toolsState: ToolsState;
+  isDebuggerPaused: boolean;
+  logCounter: number;
+  hasStaleBuildCache: boolean;
+  isRecordingScreen: boolean;
 };
 
-type ProjectStateBuildError = {
-  status: "buildError";
-  buildError: {
-    message: string;
-    platform: DevicePlatform;
-    buildType: BuildType | null;
-  };
-} & ProjectStateCommon;
+export const DeviceSessionInitialState: DeviceSessionState = {
```

Comment:
NIT: I'd prefer we'd reserve `PascalCase` for type names
```suggestion
export const DEVICE_SESSION_INITIAL_STATE: DeviceSessionState = {
```